assets/javascripts/discourse/connectors/composer-fields/persona-selector.js

```diff
@@ -0,0 +1,53 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+
+function isBotMessage(composer, currentUser) {
+  if (
+    composer &&
+    composer.targetRecipients &&
+    currentUser.ai_enabled_chat_bots
+  ) {
+    let reciepients = composer.targetRecipients.split(",");
+
+    return currentUser.ai_enabled_chat_bots.any((bot) =>
+      reciepients.any((username) => username === bot.username)
+    );
+  }
+  return false;
+}
+
+export default class BotSelector extends Component {
+  static shouldRender(args, container) {
+    return (
+      container?.currentUser?.ai_enabled_personas &&
+      isBotMessage(args.model, container.currentUser)
+    );
+  }
+
+  @service currentUser;
+
+  get composer() {
+    return this.args?.outletArgs?.model;
+  }
+
+  get botOptions() {
+    if (this.currentUser.ai_enabled_personas) {
```

Comment:
Easier just to do:


```javascript
return (this.currentUser.ai_enabled_personas || []).map() ...
```

Reply:
kind of changes behavior though (no longer returns an undefined)

Reply:
Yeah fair enough :+1: